sonar-db/src/main/java/org/sonar/db/version/v53/FixMsSqlCollation.java

```diff
@@ -0,0 +1,553 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.db.version.v53;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.db.Database;
+import org.sonar.db.dialect.MsSql;
+import org.sonar.db.version.AlterColumnsBuilder;
+import org.sonar.db.version.DdlChange;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+import static org.sonar.db.version.ColumnDefValidation.validateColumnName;
+import static org.sonar.db.version.TextColumnDef.newTextColumnDefBuilder;
+import static org.sonar.db.version.VarcharColumnDef.newVarcharColumnDefBuilder;
+
+/**
+ * Fix the collation of all columns on MsSQL
+ */
+public class FixMsSqlCollation extends DdlChange {
+
+  private static final Logger LOGGER = Loggers.get(FixMsSqlCollation.class);
+
+  private final Database db;
+
+  public FixMsSqlCollation(Database db) {
+    super(db);
+    this.db = db;
+  }
+
+  @Override
+  public void execute(Context context) throws SQLException {
+    if (!getDatabase().getDialect().getId().equals(MsSql.ID)) {
+      return;
+    }
+
+    // characteristics
+    new UpdateTableCollation(context, db, "characteristics")
+      .addVarcharColumn("kee", 100)
+      .addVarcharColumn("name", 100)
+      .addVarcharColumn("function_key", 100)
+      .addVarcharColumn("factor_unit", 100)
+      .addVarcharColumn("offset_unit", 100)
+      .execute();
+
+    // rules_parameters
+    new UpdateTableCollation(context, db, "rules_parameters")
+      .addNotNullableVarcharColumn("name", 128)
+      .addNotNullableVarcharColumn("param_type", 512)
+      .addVarcharColumn("default_value", 4000)
+      .addVarcharColumn("description", 4000)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "description" 14 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1192)